waspc/cli/src/Wasp/Cli/Command/Deps.hs

```diff
@@ -3,43 +3,52 @@ module Wasp.Cli.Command.Deps
   )
 where
 
+import Control.Monad.Except (throwError)
 import Control.Monad.IO.Class (liftIO)
 import qualified Wasp.AppSpec.App.Dependency as AS.Dependency
-import Wasp.Cli.Command (Command)
+import Wasp.Cli.Command (Command, CommandError (..))
+import Wasp.Cli.Command.Common (findWaspProjectRootDirFromCwd)
+import Wasp.Cli.Command.Compile (compileOptions)
 import Wasp.Cli.Terminal (title)
 import qualified Wasp.Generator.NpmDependencies as N
 import qualified Wasp.Generator.ServerGenerator as ServerGenerator
 import qualified Wasp.Generator.WebAppGenerator as WebAppGenerator
+import Wasp.Lib (makeAppSpec)
 import qualified Wasp.Util.Terminal as Term
 
 deps :: Command ()
-deps =
-  liftIO $
-    putStrLn $
-      unlines $
-        [ "",
-          title "Below are listed the dependencies that Wasp uses in your project. You can import and use these directly in the code as if you specified them yourself, but you can't change their versions.",
-          ""
-        ]
-          ++ printDeps
-            "Server dependencies:"
-            ( N.waspDependencies ServerGenerator.npmDepsForWasp
-            )
-          ++ [""]
-          ++ printDeps
-            "Server devDependencies:"
-            ( N.waspDevDependencies ServerGenerator.npmDepsForWasp
-            )
-          ++ [""]
-          ++ printDeps
-            "Webapp dependencies:"
-            ( N.waspDependencies WebAppGenerator.npmDepsForWasp
-            )
-          ++ [""]
-          ++ printDeps
-            "Webapp devDependencies:"
-            ( N.waspDevDependencies WebAppGenerator.npmDepsForWasp
-            )
+deps = do
+  waspProjectDir <- findWaspProjectRootDirFromCwd
+  maybeAppSpec <- liftIO $ makeAppSpec waspProjectDir (compileOptions waspProjectDir)
```

Comment:
Great points all around, and thanks for the sample code! I just had to change `id` to `pure` so I would get a `Command AppSpec` in both cases, but otherwise I really like it, thank you!!

Reply:
Ah yes I knew I must have missed something -> great to hear it works though!